app/services/qa/linked_data/authority_url_service.rb

```diff
@@ -2,59 +2,47 @@
-      # @param authority [Symbol] name of a registered authority
-      # @param subauthority [String] name of a subauthority
-      # @param action [Symbol] action with valid values :search or :term
-      # @param action_request [String] the request the user is making of the authority (e.g. query text or term id/uri)
-      # @param substitutions [Hash] variable-value pairs to substitute into the URL template
-      # @return a valid URL the submits the action request to the external authority
-      def self.build_url(action_config:, action:, action_request:, substitutions: {}, subauthority: nil)
-        action_validation(action)
-        url_config = Qa::IriTemplate::UrlConfig.new(action_url(action_config, action))
-        selected_substitutions = url_config.extract_substitutions(substitutions)
-        selected_substitutions[action_request_variable(action_config, action)] = action_request
-        selected_substitutions[action_subauth_variable(action_config, action)] = action_subauth_variable_value(action_config, subauthority, action) if subauthority.present?
+      class << self
+        # Build a url for an authority/subauthority for the specified action.
+        # @param authority [Symbol] name of a registered authority
+        # @param subauthority [String] name of a subauthority
+        # @param action [Symbol] action with valid values :search or :term
+        # @param action_request [String] the request the user is making of the authority (e.g. query text or term id/uri)
+        # @param substitutions [Hash] variable-value pairs to substitute into the URL template
+        # @return a valid URL the submits the action request to the external authority
+        def build_url(action_config:, action:, action_request:, substitutions: {}, subauthority: nil)
+          action_validation(action)
+          url_config = action_config.url_config
+          selected_substitutions = url_config.extract_substitutions(combined_substitutions(action_config, action, action_request, substitutions, subauthority))
+          Qa::IriTemplateService.build_url(url_config: url_config, substitutions: selected_substitutions)
+        end
 
-        Qa::IriTemplateService.build_url(url_config: url_config, substitutions: selected_substitutions)
-      end
+        private
 
-      def self.action_validation(action)
-        return if [:search, :term].include? action
-        raise Qa::UnsupportedAction, "#{action} Not Supported - Action must be one of the supported actions (e.g. :term, :search)"
-      end
-      private_class_method :action_validation
+          def action_validation(action)
+            return if [:search, :term].include? action
+            raise Qa::UnsupportedAction, "#{action} Not Supported - Action must be one of the supported actions (e.g. :term, :search)"
+          end
 
-      # TODO: elr - rename search and term config methods to be the same to avoid all the ternary checks
-      def self.action_url(auth_config, action)
```

Comment:
Removed... gets action_config.url_config (instance of IriTemplate::UrlConfig) instead of the url template.